Docker image install instructions added (#77)
R2Northstar/NorthstarWiki: docs/hosting-a-server-with-northstar/guides/pg9182's-Docker-Installation-Guide.md

> ---

Reviewer: Because gitbook has some weird requirements, basically the first line should be the title of the page as a top header (i.e. single `#`) which should also be the same as the filename to some degree.

All headers, should be `##` or smaller cause GitBook™ ¯\\\_(ツ)\_/¯

description: pg9182 has provided a complete docker image to run on Linux servers. This provides some server management tools and offers better performance. No physical GPU is needed. His github page can be found [here](https://github.com/pg9182/northstar-dedicated) This page is to provide basic instructions, more information and config options are found at pg9182's github. 
---

The aim of this guide is to show the steps to build a docker-compose container that starts on boot and to show basic configuration

# <a name="Prerequisites">Prerequisites</a>

| Requirement       | Description                                                                                        
| ------------------| ---------------------------------------------------------| 

> | Kernal            | Linux 5.3+, but 4.9+ should work                         |

Reviewer: ```suggestion
| Kernel            | Linux 5.3+, but 4.9+ should work                         |
```

Typo :P

| CPU               | x86_64, at least 3 cores minimum                         |
| RAM               | 2GB (Typically peaks at 1.6GB)                           |
| Disk              | 5GB                                                      |
| Network           | Recommended at least 7-20Mbps up                         |
| Docker            | Have Docker and Docker-compose installed on the machine  |

# <a name="Installation">Installation</a>

## Prep Titanfall Server files

> You will need to copy over your titanfall installation folder and place it on the Linux machine. Most of these files are for SP and therefore we can delete from the installation folder and prune the file size down to ~5GB. 

Reviewer: ```suggestion
You will need to copy over your Titanfall installation folder and place it on the Linux machine. Most of these files are for SP and therefore we can delete from the installation folder and prune the file size down to ~5GB. 
```

If we wanna be 100% correct, "Titanfall" should always be capitalised. Maybe even use "`Titanfall2` installation folder" ^^

Easiest way to do this is copy the entire titanfall folder to your desktop and delete the [following files](https://github.com/pg9182/northstar-dedicated#reducing-the-size-). 

## Copy Titanfall Folder over to the Linux Machine

> 1. FTP into your linux machine using [Filezilla](https://filezilla-project.org/) or [WinSCP](https://winscp.net/eng/download.php) 

Reviewer: This is useful for Linux noobs who may have never used scp, ssh, or the likes but I'd maybe keep it a bit more general. Maybe something like:

```suggestion
1. Access the file system of your Linux machine using tools like [Filezilla](https://filezilla-project.org/) or [WinSCP](https://winscp.net/eng/download.php).
```

> 2. Navigate to /mnt/titanfall (create directory if does not exist)

Reviewer: ```suggestion
2. Navigate to `/mnt/titanfall` (create directory if does not exist)
```

3. Copy the newly pruned titanfall folder to the server. 

![/mnt/titanfall/](https://i.postimg.cc/15HbbzFr/image.pnghttps://i.postimg.cc/15HbbzFr/image.png)

## Copy mods to /mnt/mods

> If you have configured some mods,these can be placed at /mnt/mods like below

Reviewer: ```suggestion
If you have configured some mods,these can be placed at `/mnt/mods` like below
```

![/mnt/mods](https://i.postimg.cc/tRD5jnrJ/image.png)

> 

Reviewer 2: pg9182 has a script located [here](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) that will do all this for you, will make this section a lot easier for the user.

Author: Is there any docs on how pg9182' script works? I'm looking at the code and it looks like it downloads 2.0.11.0-dedicated-mp-vpkoptim.430d3bb. Is this a prepackaged Titanfall installation?

Reviewer 2: Correct, this will download all the files that are required to run a server (instead of downloading the entire game). There's no docs since its just a simple script to download.

Author: Interesting, I'll have a play I didn't know that script existed, I'm a little concerned with the legality if I included instructions on that..

Reviewer 2: @pg9182 can probably answer that question for you.

## Create docker-compose file
We are going to be using Docker-Compose to set up our container, this gives us much more flexibility and allows us to make changes to the start up args much cleaner. 
 ```
 mkdir Titanfall
 nano docker-compose.yml
 ```

 Example docker-compose.yml

>  ```ruby

Reviewer: Does gitbook not support YAML format? I.e.
```suggestion
 ```yaml
```

 version: '3'
 services:
  northstar1: 
    image: ghcr.io/pg9182/northstar-dedicated:1-tf2.0.11.0 
    pull_policy: always 
    environment:
      - NS_PORT=37015
      - NS_PORT_AUTH=8081
      - 'NS_SERVER_NAME=[Region]Enter Server Name here'
      - 'NS_SERVER_DESC=Enter your description here'
      - |
        NS_EXTRA_ARGUMENTS=
        +setplaylist aitdm 
        +mp_gamemode aitdm 
        +map mp_angel_city
        +ns_private_match_countdown_length 0
        +ns_should_return_to_lobby 0
        +net_compresspackets_minsize 64
        +net_compresspackets 1
        +spewlog_enable 0
        +sv_maxrate 127000
    volumes:
      - /mnt/titanfall:/mnt/titanfall:ro
      - /mnt/mods:/mnt/mods:ro
    ports:
      - '37015:37015/udp'
      - '8081:8081/tcp'
    restart: always
 ```

>  A list of all the CONVARs are [here](https://r2northstar.gitbook.io/r2northstar-wiki/hosting-a-server-with-northstar/basic-listen-server#server-configuration)

Reviewer: Should be a relative link, so

```suggestion
 A list of all the CONVARs are [here](../../basic-listen-server#server-configuration)
```

Author: Thanks for the comments and suggestion, I'll put those through and PR again for review tonight.
